src/test/java/com/inventorsoft/junit/mapper/BookMapperTest.java

```diff
@@ -0,0 +1,185 @@
+        when(authorMapper.mapEntityToResponse(book.getAuthor())).thenReturn(authorResponse);
+        BookResponse givenResult = bookMapper.mapEntityToResponse(book);
+
+        assertEquals(bookResponse.getId(), givenResult.getId());
+        assertEquals(bookResponse.getTitle(), givenResult.getTitle());
+        assertEquals(bookResponse.getReleaseDate(), givenResult.getReleaseDate());
+        assertEquals(bookResponse.getDescription(), givenResult.getDescription());
+        assertEquals(bookResponse.getAuthor(), givenResult.getAuthor());
+    }
+
+    @Test
+    public void mapEntitiesToResponses() {
+        List<Book> books = List.of(book);
+        List<BookResponse> expectedResults = List.of(bookResponse);
+
+        when(authorMapper.mapEntityToResponse(book.getAuthor())).thenReturn(authorResponse);
+        List<BookResponse> givenResults = bookMapper.mapEntitiesToResponses(books);
+
+        assertEquals(expectedResults.size(), givenResults.size());
+        IntStream.range(0, expectedResults.size()).forEachOrdered(i -> {
+            assertEquals(expectedResults.get(i).getId(), givenResults.get(i).getId());
+            assertEquals(expectedResults.get(i).getDescription(), givenResults.get(i).getDescription());
+            assertEquals(expectedResults.get(i).getReleaseDate(), givenResults.get(i).getReleaseDate());
+            assertEquals(expectedResults.get(i).getTitle(), givenResults.get(i).getTitle());
+            assertEquals(expectedResults.get(i).getAuthor().getId(), givenResults.get(i).getAuthor().getId());
+            assertEquals(expectedResults.get(i).getAuthor().getName(), givenResults.get(i).getAuthor().getName());
+            assertEquals(expectedResults.get(i).getAuthor().getBirthDate(), givenResults.get(i).getAuthor().getBirthDate());
+        });
+    }
+
+    @Test
+    public void mapCreateRequestToEntity() {
+        CreateBookRequest createBookRequest = new CreateBookRequest();
+        createBookRequest.setTitle(BOOK_TITLE);
+        createBookRequest.setDescription(BOOK_DESCRIPTION);
+        createBookRequest.setReleaseDate(BOOK_RELEASE_DATE);
+        createBookRequest.setAuthorId(AUTHOR_ID);
+
+        when(authorRepository.findById(createBookRequest.getAuthorId())).thenReturn(Optional.of(author));
+        Book givenResult = bookMapper.mapCreateRequestToEntity(createBookRequest);
+
+        assertEquals(book.getId(), givenResult.getId());
+        assertEquals(book.getTitle(), givenResult.getTitle());
+        assertEquals(book.getDescription(), givenResult.getDescription());
+        assertEquals(book.getReleaseDate(), givenResult.getReleaseDate());
+        assertEquals(book.getAuthor(), givenResult.getAuthor());
+    }
+
+    @Test
+    public void whenAuthorIdIsNull_thenExceptionIsThrown() {
+        assertThrows(RuntimeException.class,
+                () -> authorRepository.findById(null)
+                        .orElseThrow(RuntimeException::new));
+    }
```

Comment:
this test name does not show which method it tests, for example, `findByIdShouldThrownExeptionWhenNotFound`,
and this test does not test anything, if you run it alone it will not add code coverage %. 
you invoke here `authorRepository.findById` as a tested method, but you test `bookMapper`, in that mapper there is no such method, that method is inside `mapCreateRequestToEntity`, so you have to test that condition there..

Create test with name `mapCreateRequestToEntityShouldThrowRuntimeExceptionWhenAuthorIsNotFound`, body should be something like:
```suggestion
      when(authorRepository.findById(id)).thenReturn(Optional.empty());
      assertThrows(RuntimeException.class, () -> bookMapper.mapCreateRequestToEntity(createBookRequest));
```